WordPress/Classes/Categories/NSURLRequest+Helpers.m

```diff
@@ -0,0 +1,61 @@
+#import "NSURLRequest+Helpers.h"
+#import "NSString+Helpers.h"
+
+@implementation NSURLRequest (Helpers)
+
++ (NSURLRequest *)requestWithURL:(NSURL *)url userAgent:(NSString *)userAgent
+{
+    NSParameterAssert(url);
+    
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
+    request.cachePolicy = NSURLRequestReturnCacheDataElseLoad;
+    
+    if (userAgent) {
+        [request setValue:userAgent forHTTPHeaderField:@"User-Agent"];
+    }
+    
+    return request;
+}
+
++ (NSURLRequest *)requestForAuthenticationWithURL:(NSURL *)loginUrl
+                                      redirectURL:(NSURL *)redirectURL
+                                         username:(NSString *)username
+                                         password:(NSString *)password
+                                      bearerToken:(NSString *)bearerToken
+                                        userAgent:(NSString *)userAgent
+{
+    NSParameterAssert(loginUrl);
+    NSParameterAssert(redirectURL);
+    NSParameterAssert(username);
+    NSParameterAssert(password != nil || bearerToken != nil);
+    
+    NSMutableURLRequest *request = [[self requestWithURL:loginUrl userAgent:userAgent] mutableCopy];
```

Comment:
You're indeed right, i meant to change that down the road, but obviously forgot to address it. Will update soon!